client/views/omnichannel/directory/calls/CallTable.tsx

```diff
@@ -0,0 +1,161 @@
+
+import GenericTable from '../../../../components/GenericTable';
+import { useRoute } from '../../../../contexts/RouterContext';
+import { useTranslation } from '../../../../contexts/TranslationContext';
+import { useEndpointData } from '../../../../hooks/useEndpointData';
+
+const useQuery = (
+	{
+		text,
+		itemsPerPage,
+		current,
+	}: {
+		text?: string;
+		itemsPerPage?: 25 | 50 | 100;
+		current?: number;
+	},
+	[column, direction]: string[],
+	userIdLoggedIn: string | null,
+): {
+	sort: string;
+	open: boolean;
+	roomName: string;
+	agents: string[];
+	count?: number;
+	current?: number;
+} =>
+	useMemo(
+		() => ({
+			sort: JSON.stringify({ [column]: direction === 'asc' ? 1 : -1 }),
+			open: false,
+			roomName: text || '',
+			agents: userIdLoggedIn ? [userIdLoggedIn] : [],
+			...(itemsPerPage && { count: itemsPerPage }),
+			...(current && { offset: current }),
+		}),
+		[column, current, direction, itemsPerPage, userIdLoggedIn, text],
+	);
+
+const CallTable: FC = () => {
+	const [params, setParams] = useState<{ text?: string; current?: number; itemsPerPage?: 25 | 50 | 100 }>({
+		text: '',
+		current: 0,
+		itemsPerPage: 25,
+	});
+	const [sort, setSort] = useState<[string, 'asc' | 'desc']>(['closedAt', 'desc']);
+	const t = useTranslation();
+	const debouncedParams = useDebouncedValue(params, 500);
+	const debouncedSort = useDebouncedValue(sort, 500);
+	const userIdLoggedIn = Meteor.userId();
+	const query = useQuery(debouncedParams, debouncedSort, userIdLoggedIn);
+	const directoryRoute = useRoute('omnichannel-directory');
+
+	const onHeaderClick = useMutableCallback((id) => {
+		const [sortBy, sortDirection] = sort;
+
+		if (sortBy === id) {
+			setSort([id, sortDirection === 'asc' ? 'desc' : 'asc']);
+			return;
+		}
+		setSort([id, 'asc']);
+	});
+
+	const onRowClick = useMutableCallback((id, token) => {
+		directoryRoute.push(
+			{
+				page: 'calls',
+				bar: 'info',
+				id,
+			},
+			{ token },
+		);
+	});
+
+	const { value: data } = useEndpointData('voip/rooms', query);
+
+	const header = useMemo(
+		() =>
+			[
+				<GenericTable.HeaderCell
+					key={'fname'}
+					direction={sort[1]}
+					active={sort[0] === 'fname'}
+					onClick={onHeaderClick}
+					sort='fname'
+					w='x400'
+				>
+					{t('Contact_Name')}
+				</GenericTable.HeaderCell>,
+				<GenericTable.HeaderCell
+					key={'phone'}
+					direction={sort[1]}
+					active={sort[0] === 'phone'}
+					onClick={onHeaderClick}
+					sort='phone'
+					w='x200'
+				>
+					{t('Phone')}
+				</GenericTable.HeaderCell>,
+				<GenericTable.HeaderCell key={'queue'} direction={sort[1]} active={sort[0] === 'queue'} onClick={onHeaderClick} sort='ts' w='x100'>
+					{t('Queue')}
+				</GenericTable.HeaderCell>,
+				<GenericTable.HeaderCell key={'ts'} direction={sort[1]} active={sort[0] === 'ts'} onClick={onHeaderClick} sort='ts' w='x200'>
+					{t('Started_At')}
+				</GenericTable.HeaderCell>,
+				<GenericTable.HeaderCell
+					key={'talkTime'}
```

Comment:
Here, we do not have "talkTime", we do have "metrics.callDuration" which is the number of millis that the call lasted 👀

Reply:
Pls ignore this 🤡